### ISSUE TITLE
`no-unbound-method` in Jasmine spy have been called test

### ISSUE THREAD

**Issue by piotrgajow** [palantir/tslint#3755]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.5.3
- __Running TSLint via__: Node.js API

#### TypeScript code being linted

```ts
it('on destroy should unsubscribe timer', () => {
    component.timerSubscription = jasmine.createSpyObj('Subscription', ['unsubscribe']);

    component.ngOnDestroy();

    expect(component.timerSubscription.unsubscribe).toHaveBeenCalled();
});
```

with `tslint.json` configuration:

```json
{
    "rulesDirectory": [
        "node_modules/codelyzer"
    ],
    "rules": {
        "no-unbound-method": true
    }
}
```

#### Actual behavior

tslint reports error in the expect line on `component.timerSubscription.unsubscribe` - `Avoid referencing unbound methods which may cause unintentional scoping of 'this'`.

#### Expected behavior

This only references the method to check if it has been called - it does not call it, nor it saves it into variable for later calls - so it should not report any errors.

**Comment by grimly** [comment]
The same happen on type detection.

This code throws a `no-unbound-method` error

```typescript
const property = Object.getOwnPropertyDescriptor(target, propertyName);
if (typeof property === 'object' && typeof property.set === 'function') {
  // ...
}
```

**Comment by maxi7587** [comment]
Have the same issue, how do you use expect() actually with tslint 5.9.1 in jasmine?

**Comment by Plondrein** [comment]
any update on this?

**Comment by borriej** [comment]
I have the same issues since I updated packages.json
```
    "@angular-devkit/build-angular": "^0.7.5",
    "@angular/cli": "^6.2.8",
    "@angular/compiler-cli": "^6.1.10",
    "@angular/language-service": "^6.1.10",
    "@types/jasmine": "^2.8.14",
    "@types/jasminewd2": "^2.0.6",
    "@types/node": "~8.9.4",
    "codelyzer": "^4.5.0",
    "jasmine-core": "~2.99.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~1.7.1",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "^2.0.4",
    "karma-jasmine": "~1.1.1",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "^5.4.1",
    "ts-node": "~5.0.1",
    "tslint": "^5.12.0",
    "typescript": "^2.9.2"
```

**Comment by piotrgajow** [comment]
I will try to work on this issue and prepare a PR, but this will be my first contribution here, so it might take some time :). If you have any tips on where to start, please let me know.

**Comment by JoshuaKGoldberg** [comment]
Awesome, thanks @piotrgajow! https://palantir.github.io/tslint/develop/custom-rules/walker-design.html is a good starting place. https://astexplorer.net with language set to TypeScript is really useful to explore what the AST looks like for code snippets.

Feel free to ask for help on Gitter or Twitter if you have any questions! The docs are a little old and you definitely wouldn't be the only one to want to ask for clarifications!

**Comment by JoshuaKGoldberg** [comment]
Great question... we can't break backwards compatibility for the rule, so it should always allow something like `"no-unbound-method": [true, "ignore-static"]`. How about allowing either a string or the object as the parameter? Does that look doable with the metadata format?

**Comment by JoshuaKGoldberg** [comment]
Fixed by #4472! 🙌
